src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -126,7 +152,9 @@ class GcsActor {
   /// Get the `Address` of this actor.
   const rpc::Address &GetAddress() const;
 
-  /// Update the state of this actor.
+  /// Update the state of this actor and refreshes metrics. Do not update the
+  /// state of the underlying proto directly via set_state(), otherwise metrics
```

Comment:
should we just disallow `mutable_*` API?

Reply:
It's quite a big refactor. I don't think it's worth doing in this PR since it only pertains to the actor state.